src/clients/apify-client.ts

```diff
@@ -178,85 +178,54 @@ export class ExtApifyClient extends ApifyClient implements ExtendedApifyClient {
-                                callback({ run: undefined, error: e as Error }),
-                            );
-                        }
-                    } else {
-                        this.context.logger.error("Something wrong with the Apify orchestrator's queue!");
-                    }
-                } else if (!this.retryOnInsufficientResources) {
-                    const runRequest = this.runRequestsQueue.dequeue();
-                    if (!runRequest) {
-                        throw new Error('Insufficient resources have been retrieved but no runRequest found!');
+
+                const { runName, input, options } = nextRunRequest;
+
+                this.context.logger.prfxInfo(runName, 'Starting next', { queue: this.runRequestsQueue.length });
+
+                let result: RunResult;
+
+                try {
+                    const run = await nextRunRequest.startRun(input, options);
+                    result = { run, error: undefined };
+                } catch (startError) {
+                    this.context.logger.prfxError(runName, 'Failed to start Run', {
+                        message: (startError as Error)?.message,
+                    });
+                    const error = await parseStartRunError(startError, runName, getRequiredMemoryMbytes);
+                    result = { run: undefined, error };
+                }
+
+                const { run, error } = result;
+
+                if (run) {
+                    await this.context.runsTracker.updateRun(runName, run);
+                    for (const callback of nextRunRequest.startCallbacks) {
+                        callback({ run, error: undefined });
                     }
-                    const { runName } = runRequest;
-
-                    const errorToThrow = (() => {
-                        if (!hasEnoughMemory) {
-                            return new InsufficientMemoryError(runName, requiredMemoryGBs, availableMemoryGBs);
-                        }
-                        if (!canRunMoreActors) {
-                            return new InsufficientActorJobsError(runName);
-                        }
-                        throw new Error(
-                            'Insufficient resources have been retrieved but they did not match any of the checks!',
-                        );
-                    })();
-
-                    this.context.logger.prfxError(
-                        runName,
-                        'Failed to start Run and retryOnInsufficientResources is set to false',
-                        {
-                            message: errorToThrow.message,
-                        },
-                    );
-                    runRequest.startCallbacks.map((callback) => callback({ run: undefined, error: errorToThrow }));
-                } else {
-                    // Wait for sometime before checking again
+                } else if (
+                    this.retryOnInsufficientResources &&
```

Comment:
I know this isn't related to this PR but was there ever a use-case for not retrying? Feels like an option we preemptively added but now might keep with us but perhaps I'm wrong :)